packages/shopping/src/__tests__/acceptance/user.controller.acceptance.ts

```diff
@@ -238,21 +241,33 @@ describe('UserController', () => {
 
   describe('user product recommendation (service) api', () => {
     let recommendationService: HttpServer;
+    let recommendationGRPCService: Server;
 
     before(async () => {
       recommendationService = createRecommendationServer();
       await recommendationService.start();
+      recommendationGRPCService = createGRPCRecommendationServer();
+      recommendationGRPCService.start();
     });
 
     after(async () => {
       await recommendationService.stop();
+      recommendationGRPCService.forceShutdown();
     });
 
     it('returns product recommendations for a user', async () => {
       const newUser = await createAUser();
       await client
         .get(`/users/${newUser.id}/recommend`)
-        .expect(200, recommendations);
+        .expect(200, recommendations['user001']);
+    });
+
+    it('returns product recommendations for a user using gRPC', async () => {
+      const recommender = await app.get<RecommenderService>(
+        'services.RecommenderGrpcService',
+      );
+      const products = await recommender.getProductRecommendations('user001');
```

Comment:
Can we have a test case for the user not in recommendations such as `user003`?

Reply:
The recommender is a mock service that returns same products for users other than `user002`

Reply:
I added one for user004.